models/index_model.py

```diff
@@ -35,15 +36,14 @@
 SHORT_TO_LONG_CACHE = {}
 
 
-def get_and_query(user_id, index_storage, query, llm_predictor):
-    # TODO Do prediction here for token usage
+def get_and_query(user_id, index_storage, query, response_mode, llm_predictor):
     index: [GPTSimpleVectorIndex, ComposableGraph] = index_storage[
         user_id
     ].get_index_or_throw()
     if isinstance(index, GPTTreeIndex):
-        response = index.query(query, verbose=True, child_branch_factor=2)
+        response = index.query(query, verbose=True, child_branch_factor=2, llm_predictor=llm_predictor)
```

Comment:
Can you add the corresponding calls to the usage service to update the actual usage whenever we call a `.query` and when we construct GPTTreeIndexes for Deep Compose?

Reply:
Already counts when the query command is called, added counting to deep compose too